src/Adapter/Order/Refund/OrderRefundCalculator.php

```diff
@@ -81,49 +81,58 @@ public function computeOrderRefund(
             $orderDetailList,
             $precision
         );
-        $refundedAmount = 0;
+
+        $refundedAmount = new Number('0');
         foreach ($productRefunds as $orderDetailId => $productRefund) {
-            $refundedAmount += $productRefund['amount'];
+            $refundedAmount = $refundedAmount->plus(new Number((string) $productRefund['amount']));
         }
 
         $voucherChosen = false;
-        $voucherAmount = 0;
+        $voucherAmount = new Number('0');
         if ($voucherRefundType === VoucherRefundType::PRODUCT_PRICES_EXCLUDING_VOUCHER_REFUND) {
-            //@todo: Check if it matches order_discount_price in legacy
-            $refundedAmount -= $voucherAmount = (float) $order->total_discounts;
+            $voucherAmount = new Number((string) $order->total_discounts);
+            $refundedAmount = $refundedAmount->minus($voucherAmount);
         } elseif ($voucherRefundType === VoucherRefundType::SPECIFIC_AMOUNT_REFUND) {
             $voucherChosen = true;
             $refundedAmount = $voucherAmount = $chosenVoucherAmount;
         }
 
-        $shippingCostAmount = (float) (string) $shippingRefund;
-        if ($shippingCostAmount > 0) {
-            $shippingMaxRefund = $isTaxIncluded ? $order->total_shipping_tax_incl : $order->total_shipping_tax_excl;
+        $shippingCostAmount = new Number((string) ($shippingRefund ?? 0));
```

Comment:
```suggestion
        $shippingCostAmount = $shippingRefund ?? new Number('0');
```

Since `$shippingRefund` is already a Number (and Number is immutable)

Reply:
:+1: